cardano-cli/src/Cardano/CLI/EraBased/Options/Query.hs

```diff
@@ -124,13 +124,14 @@ pQueryProtocolParametersCmd envCli =
       <*> pNetworkId envCli
       <*> pMaybeOutputFile
 
-pQueryTipCmd :: EnvCli -> Parser (QueryCmds era)
-pQueryTipCmd envCli =
+pQueryTipCmd :: CardanoEra era -> EnvCli -> Parser (QueryCmds era)
+pQueryTipCmd era envCli =
   fmap QueryTipCmd $
     QueryTipCmdArgs
       <$> pSocketPath envCli
       <*> pConsensusModeParams
       <*> pNetworkId envCli
+      <*> pTarget era
```

Comment:
Then what you can do here is use `maybe` and `forEraMaybeEon` so that when we cannot get the `ConwayEraOnwards` witness, we default to `VolatileTip`.

Reply:
Shouldn't that go inside the `pTarget` as, in my opinion, the defaulting is its implementation detail?